### PR TITLE
[FLINK-4935] [webfrontend] Submit job with savepoint

### PR DESCRIPTION
**Pull request by uce**
This builds on top of #2712 and only the last commit 4265834 is relevant.

Allows to submit a job via the REST API and restoring from a savpeoint:

screen shot 2016-10-27 at 16 52 00
## Changes
- Adds documentation for the REST API `/jars/:jarid/run` command
- Adds two new query parameters to run a JAR with the savepoint restore settings:
  1. `savepointPath`: Sets the savepoint path
  2. `ignoreUnmappedState`: Ignores unmapped state (default `false`)

### CODE REVIEW

File: flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/handlers/JarActionHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+public class JarActionHandlerTest {
+
+	/**
+	 * Test that the savepoint settings are correctly parsed.
+	 */
+	@Test
+	public void testSavepointRestoreSettings() throws Exception {
+		Map<String, String> pathParams = new HashMap<>();
+		pathParams.put("jarid", "required"); // required
+		pathParams.put("savepointPath", "ignored"); // should be ignored
```
Comment (zentol) [inline flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/handlers/JarActionHandlerTest.java]: we may want to mention that it should be ignored as it belongs into the query parameters

**Comment by zentol** [comment]
minor comments, otherwise +1

File: docs/monitoring/rest_api.md [hunk 2]
```diff
@@ -652,3 +657,32 @@ The `savepointPath` points to the external path of the savepoint, which can be u
...
+If the call succeeds, you will get a response with the ID of the submitted job.
+
+**Example:** Run program with a savepoint
+
+~~~
+POST: /jars/MyProgram.jar/run?savepointPath=/my-savepoints/savepoint-1bae02a80464&ignoreUnmappedState=true
+~~~
+
+~~~
+{"jobid": "869a9868d49c679e7355700e0857af85"}
```
Comment (uce) [inline docs/monitoring/rest_api.md]: That's an example response.

File: flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/handlers/JarActionHandlerTest.java (hunk 1 above)
Comment (uce) [inline flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/handlers/JarActionHandlerTest.java]: That's what the comment says. Did you mean something else?

**Comment by uce** [comment]
Thanks for the review. I would like to merge this as soon as #2712 is in.

**Pull request closed without merging**